src/main.c

```diff
@@ -30,6 +31,11 @@ LOG_MODULE_REGISTER(golioth_rd_template, LOG_LEVEL_DBG);
 #include <modem/modem_info.h>
 #endif
 
+/* Macrobatics to convert APP_BUILD_VERSION to a string */
+#define _STR(s)			 #s
+#define STR(s)			 _STR(s)
+#define APP_BUILD_VERSION_STRING STR(APP_BUILD_VERSION)
+
```

Comment:
I'm not sure why `APP_VERSION_STRING` is quoted in `app_version.h` but `APP_BUILD_VERSION` is not... (see https://docs.zephyrproject.org/latest/build/version/index.html#use-in-code for details)

It feels like there should be a better way to print the `APP_BUILD_VERSION` as a string, but I'm not aware of one.